app/utils/database.py

```diff
@@ -360,7 +368,7 @@ def authenticate_user(email, password):
         # attacks
         check_password_hash("This-always-fails", password)
     logging.info("Failed login attempt for %s", email)
-    return None
+    return User()
```

Comment:
Why not return a None here? What does this dummy User contain, every field initialized to a default null value? Isn't that potentially problematic?

Reply:
The idea is that an anonymous user is still a user, and I figured that it would allow the rest of the code to be cleaner if you could trust that the user was always the same type. Not sure it's a good idea, but that was the reasoning!